README.md

```diff
@@ -1 +1,43 @@
-# parse-semistruct
\ No newline at end of file
+# Welcome!
+
+`semistruct-parse` is a project providing a parser for semistructured lines of
+text. These are primarily used in a custom journald logging driver for the
+purpose of feeding semistructured data into journald from docker.
+
+## Testing
+
+You need the following dependencies in order to run the test suite:
+
+- `github.com/leanovate/gopter`
+- `github.com/andyleap/parser`
+
+There is a small hand-ful of hand-written unit tests for the log lines we
+expected to be emitting and then a suite of property tests for each field of the
+semistructured log line, using `gopter` for the framework.
+
+## The Format
+
+Here is an example of a semistructured line that, when parsed by the top-level
+`ParseSemistruct` combinator, emits a struct of standard Go data structures
+representing the three primary fields:
+
+`!< 8 [db3:slave] { region="us-east-d z2" uuid=3c79bfc0-4b27-4c99-a952-8f435bbf0514 } >!`
```

Comment:
this spec looks wrong. The key value pairs should conform to well formed JSON rules.

```
journalctl -o json

{ "__CURSOR" : "s=a77347956c9043f5b4dde06181d611bf;i=2547d;b=4727176050834387a0bd71d0b94d80f2;m=3ff3b6aee2;t=52b0c669b8a79;x=d38cfb4e25682f66", "__REALTIME_TIMESTAMP" : "1454707144624761", "__MONOTONIC_TIMESTAMP" : "274671775458", "_BOOT_ID" : "4727176050834387a0bd71d0b94d80f2", "PRIORITY" : "6", "SYSLOG_FACILITY" : "3", "CODE_FILE" : "src/core/unit.c", "CODE_LINE" : "1432", "CODE_FUNCTION" : "unit_status_log_starting_stopping_reloading", "SYSLOG_IDENTIFIER" : "systemd", "MESSAGE_ID" : "7d4958e842da4a758f6c1cdc7b36dcc5", "USER_UNIT" : "dbus.socket", "MESSAGE" : "Starting D-Bus User Message Bus Socket.", "_TRANSPORT" : "journal", "_PID" : "13949", "_UID" : "1005", "_GID" : "1006", "_COMM" : "systemd", "_EXE" : "/usr/lib/systemd/systemd", "_CMDLINE" : "/usr/lib/systemd/systemd --user", "_CAP_EFFECTIVE" : "0", "_SYSTEMD_CGROUP" : "/user.slice/user-1005.slice/user@1005.service/init.scope", "_SYSTEMD_OWNER_UID" : "1005", "_SYSTEMD_UNIT" : "user@1005.service", "_SYSTEMD_USER_UNIT" : "init.scope", "_SYSTEMD_SLICE" : "user-1005.slice", "_SOURCE_REALTIME_TIMESTAMP" : "1454707144613431", "_MACHINE_ID" : "0715c0deb7214e8ea7ca35e1d7b09ad7", "_HOSTNAME" : "fileserver" }
```